ecdsa/party/implementation.go

```diff
@@ -305,9 +313,23 @@ func (p *Impl) authoriseSignerToSign(digest Digest, signer *singleSigner) error
 	defer signer.mtx.Unlock()
 
 	if signer.localParty == nil {
+		randomnessSeed := append(p.partiesSharedSecret, digest[:]...)
+		parties, err := shuffleParties(randomnessSeed, p.parameters.Parties().IDs())
+		if err != nil {
+			return err
+		}
+
+		parties = tss.SortPartyIDs(parties[:p.parameters.Threshold()+1])
+
+		selfIdInCurrentCommittee := p.selfInSigningCommittee(parties)
+		if selfIdInCurrentCommittee == nil {
+			return ErrNotInSigningCommittee
+		}
+
 		signer.localParty = signing.NewLocalParty(
 			(&big.Int{}).SetBytes(digest[:]),
-			p.parameters,
+			// p.parameters,
+			tss.NewParameters(tss.S256(), tss.NewPeerContext(parties), selfIdInCurrentCommittee, len(parties), p.parameters.Threshold()),
```

Comment:
Why do we need to pass selfIdInCurrentCommittee? It is always true when creating a localparty, right? otherwise we wouldn't have created it

Reply:
The partID.index is different: `tss.SortPartyIDS(parties)` changes the indices of each party.  So I take the modified `partyId`.

Reply:
I don't get it still. It seems like something you could just compute rather than take as a parameter